src/frontend/http.rs

```diff
@@ -279,6 +439,57 @@ mod tests {
     const V2_ETAG: &str =
         "06d033ece6645de592db973644cf7357255f24536ff7b03c3b2ace10736f7636";
 
+    // Create a mock upload request and execute it
+    async fn mock_upload_request(
+        filename: &str,
+        mut file_content: Vec<u8>,
+        schema_name: &str,
+        table_name: &str,
+        schema_json: Option<&str>,
+        context: Arc<dyn SeafowlContext>,
+    ) -> Response<Bytes> {
+        let handler = filters(context);
+
+        let mut body: Vec<u8> = vec![];
+
+        if let Some(schema_json) = schema_json {
+            body = "--42\r\n\
+                Content-Disposition: form-data; name=\"schema\"\n\
+                Content-Type: application/json\n\n"
+                .to_string()
+                .as_bytes()
+                .to_vec();
+            body.append(&mut schema_json.as_bytes().to_vec());
+            body.append(&mut "\n\n".as_bytes().to_vec());
+        }
+
+        body.append(
+            &mut format!(
+                "--42\r\n\
+                Content-Disposition: form-data; name=\"data\"; filename=\"{}\"\n\
+                Content-Type: application/octet-stream\n\n",
+                filename
+            )
+            .as_bytes()
+            .to_vec(),
+        );
+
+        body.append(&mut file_content);
+        body.append(&mut "--42--".as_bytes().to_vec());
```

Comment:
Can you also test an upload that actually has multiple parts (i.e. more than one boundary)? It could be easier to test this with a test in `end_to_end.rs` that calls out into `curl` to make sure this works with real clients.